api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -36,6 +36,12 @@ public static NameSupport getInstance() {
 			return singleton;
 		}
 	}
+
+	public static void init() {
+		if (singleton == null) {
```

Comment:
Is this just for testing?

Reply:
This method is used to initialise first instance of NameSupport, otherwise it will throw Not Yet Instantiated"

Reply:
Let me rephrase, did you add the init method simply for tests?

Reply:
I think so, because Spring doesn't initialise this bean only in unit tests :blush:

Reply:
Did you consider putting the bean in TestingApplicationContext.xml? 😊

Reply:
So if I add something like this into TestingApplicationContext it would autowire NameTemplate?

        ```<bean id="nameSupport"
	      class="org.openmrs.layout.name.NameSupport">
	```</bean>

Reply:
@dkayiwa Could you tell me how to initialise this class instance before running any test? 😊

Reply:
I've put something like this
https://github.com/openmrs/openmrs-core/pull/2470/files#diff-de63ef1337d1da40a8826bb63ee8fc92R33

Reply:
And did it work? 😊

Reply:
Unfortunately no, that's why I ask you